src/unitxt/operators.py

```diff
@@ -658,6 +665,65 @@ def process(self, stream: Stream, stream_name: str = None) -> Generator:
                 yield instance
 
 
+class FilterByListsOfValues(SingleStreamOperator):
+    """
+    Filters a stream, yielding only instances that  whose field values are included in the specified value lists.
+
+    Args:
+        values (Dict[str, Any]): For each field, the list of values that instances should match to be included in the output.
+    """
+
+    values: Dict[str, Any]
+
+    def verify(self):
+        for key, value in self.values.items():
+            if not isinstance(value, list):
+                raise ValueError(f"The filter for key ('{key}') in FilterByListsOfValues is not a list but '{value}'")
+
+    def process(self, stream: Stream, stream_name: str = None) -> Generator:
+        for instance in stream:
+            if any(instance[key] in value for key, value in self.values.items()):
+                yield instance
+
+
+class Intersect(FieldOperator):
+    """
+    Intersects the value of a field, which must be a list, with a given list
+    Args:
+        allowed_values (list) - list to intersect
+    """
+
+    allowed_values: List[Any]
+
+    def verify(self):
+        if not isinstance(self.allowed_values, list):
+            raise ValueError(f"The allowed_values is not a list but '{self.allowed_values}'")
+
+    def process_value(self, value: Any) -> Any:
+        if not isinstance(value, list):
+            raise ValueError(f"The value in field is not a list but '{value}'")
+        return [e for e in value if e in self.allowed_values]
+
+
+class RemoveValues(FieldOperator):
+    """
+    Removes elements in a field, which must be a list, using a given list of unallowed
+    Args:
+        unallowed_values (list) - removed_values
+    """
+
+    unallowed_values: List[Any]
+
+    def verify(self):
+        if not isinstance(self.unallowed_values, list):
+            raise ValueError(f"The unallowed_values is not a list but '{self.unallowed_values}'")
+
+    def process_value(self, value: Any) -> Any:
+        if not isinstance(value, list):
```

Comment:
Same comment as above

Reply:
Same as above.  Prevented (process_every_value is True) .